# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/WSRunObject.cpp
# Commit: 0475e2855e5c
# Full Hash: 0475e2855e5c4b03dfce9da151fd4ac2bb1367c3
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-05-23 15:52:15
# Regressor Bug: 1534394
# File Overlap Count: 1
# Description:
#   Bug 1534394 - Make WSRunObject::InsertText() set aPointAfterInsertedString by itself when HTMLEditor::InsertTextWithTransaction() returns error r=m_kato
#   
#   Oddly, `WSRunObject::InsertText()` returns `NS_OK` even when
#   `HTMLEditor::InsertTextWithTransaction()` returns error.  However,
#   it fails if insertion point is not editable like `<noscript>` element.
# ==============================================================================

diff -r 4cd6d838796a -r 0475e2855e5c editor/libeditor/WSRunObject.cpp
--- a/editor/libeditor/WSRunObject.cpp	Thu May 23 06:35:52 2019 +0000
+++ b/editor/libeditor/WSRunObject.cpp	Thu May 23 06:06:18 2019 +0000
@@ -368,12 +368,18 @@
     }
   }
 
-  // Ready, aim, fire!
+  // XXX If the point is not editable, InsertTextWithTransaction() returns
+  //     error, but we keep handling it.  But I think that it wastes the
+  //     runtime cost.  So, perhaps, we should return error code which couldn't
+  //     modify it and make each caller of this method decide whether it should
+  //     keep or stop handling the edit action.
   nsresult rv =
       MOZ_KnownLive(mHTMLEditor)
           ->InsertTextWithTransaction(aDocument, theString, pointToInsert,
                                       aPointAfterInsertedString);
   if (NS_WARN_IF(NS_FAILED(rv))) {
+    // XXX Temporarily, set new insertion point to the original point.
+    *aPointAfterInsertedString = pointToInsert;
     return NS_OK;
   }
   return NS_OK;